packages/design-system/src/components/TextField/Mask.jsx

```diff
@@ -69,33 +47,38 @@ function toDigitsAndAsterisks(value) {
 }
 
 /**
- * Remove all non-digits
+ * Performs various transforms to format provided string as currency.
  * @param {String} value
  * @returns {String}
  */
-function toDigits(value) {
-  return value.replace(/[^\d]/g, '');
-}
-
-/**
- * Convert string into a number (positive or negative float or integer)
- * @param {String} value
- * @returns {Number}
- */
-function toNumber(value) {
-  const sign = value.charAt(0) === '-' ? -1 : 1;
-  const parts = value.split('.');
-  // This assumes if the user adds a "." it should be a float. If we want it to
-  // evaluate as an integer if there are no digits beyond the decimal, then we
-  // can change it.
-  const hasDecimal = parts[1] !== undefined;
-  if (hasDecimal) {
-    const a = toDigits(parts[0]);
-    const b = toDigits(parts[1]);
-    return sign * parseFloat(`${a}.${b}`);
-  } else {
-    return sign * parseInt(toDigits(parts[0]));
+function toCurrency(value) {
+  // Determine if the value is positive or negative.
+  const sign = value.startsWith('-') ? '-' : '';
+  // Remove all characters except digits and decimal points.
+  value = value.replace(/[^\d.]/g, '');
+  // Remove all but the first decimal point.
+  const firstDecimalPointIndex = value.indexOf('.');
+  value = value.replace(/[.]/g, (match, index) => {
+    return index > firstDecimalPointIndex ? '' : match;
+  });
+  // Remove leading zeroes (we'll add one back later if needed).
+  value = value.replace(/^0+/g, '');
+  // Split into whole number and fractional parts based on decimal point.
+  let [whole, fractional = ''] = value.split('.');
+  if (fractional !== '') {
+    if (fractional.length === 1) {
+      // Pad with a zero for two decimal places.
+      fractional = fractional.concat('0');
+    } else if (fractional.length > 2) {
+      // Cut all characters after the second decimal place.
+      fractional = fractional.slice(0, 2);
+    }
+    // Clear the fractional if there's no cents. Add the decimal back here.
+    fractional = fractional === '00' ? '' : `.${fractional}`;
   }
+  // Add commas for readability or return zero.
+  whole = whole === '' ? '0' : whole.replace(/\B(?=(?:\d{3})+(?!\d))/g, ',');
```

Comment:
This is definitely the most complicated / illegible part.  To be brief, we're "replacing" the zero-length space between groups of 3 digits with a comma.  

A variation of this regex can be found all over the place; the only modification I made was the addition of the non-capturing group (`?:`) which just makes the match return cleaner (doesn't actually affect the replacement).

Here's a demo of it in action alongside an explanation of the tokens that may be helpful: https://regex101.com/r/JPocnt/2

Reply:
What if you actually include this link in the comment? I think it makes great documentation

Reply:
Could we maybe include this explanation in a comment here?